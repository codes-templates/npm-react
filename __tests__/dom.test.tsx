import '@testing-library/jest-dom';
import { describe, test, expect } from 'vitest';

describe('dom test', () => {
  test.concurrent('testing-library/jest-dom', async () => {
    const div = document.createElement('div');
    div.id = 'adm-mask';
    expect(div).not.toBeNull();
    expect(div).toBeDefined();
    expect(div).toBeInstanceOf(HTMLDivElement);

    await document.body.appendChild(div);

    const mask = document.body.querySelector('#adm-mask') as HTMLElement;
    expect(mask).toMatchSnapshot();
    div.remove();
    expect(mask).not.toBeInTheDocument();
  });
});
