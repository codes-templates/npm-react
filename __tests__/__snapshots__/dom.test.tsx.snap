// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`dom test > testing-library/jest-dom 1`] = `
<div
  id="adm-mask"
/>
`;
